src/Bicep.Core/TypeSystem/CyclicCheckVisitor.cs

```diff
@@ -69,6 +69,9 @@ private void VisitDeclaration<TDeclarationSyntax>(TDeclarationSyntax syntax, Act
             }
         }
 
+        public override void VisitMetadataDeclarationSyntax(MetadataDeclarationSyntax syntax)
+            => VisitDeclaration(syntax, base.VisitMetadataDeclarationSyntax);
+        
```

Comment:
Can there be cycles? I'd expect this feature to behave more like outputs, where there's no ability for a `meta` block to be able to reference another one.